src/policy/semantic.rs

```diff
@@ -78,6 +124,42 @@ impl<Pk: MiniscriptKey> ForEachKey<Pk> for Policy<Pk> {
 }
 
 impl<Pk: MiniscriptKey> Policy<Pk> {
+    fn term_name(&self) -> String {
+        match *self {
+            Policy::Threshold(k, ref subs) => {
+                if k == subs.len() {
+                    String::from(format!("and({},{})", k, subs.len()))
+                } else if k == 1 {
+                    String::from(format!("or({},{})", k, subs.len()))
+                } else {
+                    String::from(format!("thresh({},{})", k, subs.len()))
+                }
+            }
+            _ => self.to_string(),
+        }
+    }
+
+    fn _pprint_tree(&self, prefix: String, last: bool) {
+        let prefix_current = if last { "`-- " } else { "|-- " };
+
+        println!("{}{}{}", prefix, prefix_current, self.term_name());
+
+        let prefix_child = if last { "    " } else { "|   " };
+        let prefix = prefix + prefix_child;
+
+        if let Policy::Threshold(_k, ref subs) = *self {
+            let last_child = subs.len() - 1;
+
+            for (i, child) in subs.iter().enumerate() {
+                child._pprint_tree(prefix.to_string(), i == last_child);
+            }
+        }
+    }
+
+    /// Pretty Print a tree
+    pub fn pprint_tree(&self) {
```

Comment:
In 4d23b799f6d066358f2bbe3792981e43db82f417:

Can we make this `pub(crate)`? I think we can come up with a much better API here and I'd rather not expose this one.